src/server/debug/server/server.go

```diff
@@ -367,14 +370,12 @@ func collectBinary(tw *tar.Writer, prefix ...string) error {
 	}, prefix...)
 }
 
-func redirectBinaryFunc(ctx context.Context) redirectFunc {
-	return func(c debug.DebugClient, filter *debug.Filter) (io.Reader, error) {
-		binaryC, err := c.Binary(ctx, &debug.BinaryRequest{Filter: filter})
-		if err != nil {
-			return nil, errors.EnsureStack(err)
-		}
-		return grpcutil.NewStreamingBytesReader(binaryC, nil), nil
+func redirectBinaryFunc(ctx context.Context, c debug.DebugClient, filter *debug.Filter) (io.Reader, error) {
```

Comment:
Since it no longer returns a function, it should probably be renamed to `redirectBinary`.